projects/RabbitMQ.Client/client/events/AsyncEventingBasicConsumer.cs

```diff
@@ -86,5 +88,13 @@ public override async Task HandleModelShutdown(object model, ShutdownEventArgs r
                 await _shutdownWrapper.InvokeAsync(this, reason).ConfigureAwait(false);
             }
         }
+
+        private async Task BasicDeliverWrapper(Task deliverTask, BasicDeliverEventArgs eventArgs)
```

Comment:
I should change this to skip the wrapper if the Activity isn't created. That reduces the overhead a bit.

Reply:
Did a quick test. Had next to no impact, so skipping it.